src/GitHub.App/Services/GitClient.cs

```diff
@@ -96,30 +96,30 @@ public Task Fetch(IRepository repository, string remoteName)
 
         public Task Fetch(IRepository repo, UriString cloneUrl, params string[] refspecs)
         {
-            var httpsString = cloneUrl.ToRepositoryUrl().ToString();
-
             foreach (var remote in repo.Network.Remotes)
             {
-                var remoteUrl = new UriString(remote.Url);
-                if (!remoteUrl.IsHypertextTransferProtocol)
-                {
-                    // Only match http urls
-                    continue;
-                }
-
-                var remoteHttpsString = remoteUrl.ToRepositoryUrl().ToString();
-                if (remoteHttpsString.Equals(httpsString, StringComparison.OrdinalIgnoreCase))
+                if (UriString.RepositoryUrlsAreEqual(new UriString(remote.Url), cloneUrl))
                 {
-                    return Fetch(repo, defaultOriginName, refspecs);
```

Comment:
There was a bug here. This should have used `remote.Name` not `defaultOriginName`.